Gofer.NET.Tests/GivenATaskClient.cs

```diff
@@ -28,37 +29,59 @@ public async Task ItContinuesListeningWhenATaskThrowsAnException()
 
             taskClient.CancelListen();
             await task;
-            
+
+            TaskQueueTestFixture.EnsureSemaphore(semaphoreFile);
+        }
+
+        [Fact]
+        public async Task ItStopsOnCancellation()
+        {
+            var semaphoreFile = Path.GetTempFileName();
+
+            var waitTime = 2000;
+
+            var taskQueue = TaskQueueTestFixture.UniqueRedisTaskQueue();
+            var taskClient = new TaskClient(taskQueue);
+            var cancellation = new CancellationTokenSource();
+
+            await taskClient.TaskQueue.Enqueue(() => TaskQueueTestFixture.WaitForCancellationAndWriteSemaphore(semaphoreFile, default));
+
+            var task = Task.Run(async () => await taskClient.Listen(cancellation.Token), CancellationToken.None);
+            await Task.Delay(waitTime, CancellationToken.None);
+            cancellation.Cancel();
```

Comment:
Enqueued task (WaitForCancellationAndWriteSemaphore) will be canceled at this moment.